flashlight/autograd/Variable.h

```diff
@@ -116,6 +117,42 @@ class Variable {
    */
   af::array& array() const;
 
+  /**
+   * Casts the variable to the given data type.
+   *
+   * @param[in] type target data type
+   * @param[in] castGrad determines if the gradient should be casted in case it
+   * exists.
+   */
+  __forceinline void inPlaceCast(af::dtype type, bool castGrad = true) {
```

Comment:
Is there some reason to define these in the header/inline? Would prefer not to.

Reply:
Also, I'm not sure if a situation in which we'd want `castGrad` to *not* be true. If that were the case, we could just potentially set the gradient ourselves, but it seems like an edge case. I think we should remove the argument/condition.

Reply:
Done, done, rebased, and pushed.